system/Test/FeatureTestTrait.php

```diff
@@ -101,6 +101,25 @@ public function withSession(array $values = null)
 		return $this;
 	}
 
+	/**
+	 * Set request's headers
+	 * 
+	 * Example of use 
+	 * withHeaders([
+	 *  'Authorization' => 'Token'
+	 * ])
+	 * 
+	 * @param array|null Array of headers
+	 * 
+	 * @return $this
+	 */
+	public function withHeaders(array $headers = null)
```

Comment:
Maybe we can have the default value of `$headers` to an empty array `[]`, instead? With this change, we can remove the need for the not empty check in your foreach later.

Reply:
I like using empty array as the default since that is the class property default.